src/frontend/src/icp-eth/components/tokens/IcManageTokens.svelte

```diff
@@ -69,40 +70,47 @@
 			);
 
 	let tokens: IcrcCustomToken[] = [];
-	$: tokens = filteredTokens.map(({ ledgerCanisterId, enabled, ...rest }) => ({
-		ledgerCanisterId,
-		enabled: modifiedTokens[`${ledgerCanisterId}`]?.enabled ?? enabled,
+	$: tokens = filteredTokens.map(({ id, network, enabled, ...rest }) => ({
+		id,
+		network,
+		enabled: modifiedTokens[network.id]?.[id]?.enabled ?? enabled,
 		...rest
 	}));
 
 	let noTokensMatch = false;
 	$: noTokensMatch = tokens.length === 0;
 
-	let modifiedTokens: Record<CanisterIdText, IcrcCustomToken> = {};
-	const onToggle = ({
-		detail: { ledgerCanisterId, enabled, ...rest }
-	}: CustomEvent<IcrcCustomToken>) => {
-		const { [`${ledgerCanisterId}`]: current, ...tokens } = modifiedTokens;
+	let modifiedTokens: Record<NetworkId, Record<TokenId, IcrcCustomToken>> = {};
+	const onToggle = ({ detail: { id, network, ...rest } }: CustomEvent<IcrcCustomToken>) => {
+		const { id: networkId } = network;
 
-		if (nonNullish(current)) {
-			modifiedTokens = { ...tokens };
+		modifiedTokens[networkId] = modifiedTokens[networkId] || {};
+
+		if (nonNullish(modifiedTokens[networkId][id])) {
+			delete modifiedTokens[networkId][id];
+			if (Object.getOwnPropertySymbols(modifiedTokens[networkId]).length === 0) {
+				delete modifiedTokens[networkId];
+			}
 			return;
 		}
 
-		modifiedTokens = {
-			[`${ledgerCanisterId}`]: {
-				ledgerCanisterId,
-				enabled,
-				...rest
-			},
-			...tokens
-		};
+		modifiedTokens[networkId][id] = { id, network, ...rest };
 	};
 
 	let saveDisabled = true;
-	$: saveDisabled = Object.keys(modifiedTokens).length === 0;
-
-	const save = () => dispatch('icSave', Object.values(modifiedTokens));
+	$: saveDisabled = Object.getOwnPropertySymbols(modifiedTokens).every(
+		(networkId) => Object.getOwnPropertySymbols(modifiedTokens[networkId]).length === 0
+	);
+
+	const save = () =>
+		dispatch(
+			'icSave',
+			Object.getOwnPropertySymbols(modifiedTokens).flatMap((networkId) =>
```

Comment:
Using e.g. a reducer would we manage to do build the result in a single iteration?

Reply:
well, the first iteration I did was not a nested record, but

`let modifiedTokens: Record<string, IcrcCustomToken | undefined> = {};`

where the index was

`${networkId.description}-${id.description}`

a bit more verbose, but it takes one loop off, so it is preferable: changing it